app/updatemanager.go

```diff
@@ -396,6 +397,33 @@ func (c *ControlMapPool) QueryAndUpdate(state string) (action string) {
 	return "continue"
 }
 
+// TODO - rename to NextUpdateControlMapTime / or similar
+func (c *ControlMapPool) NextControlMapHalfTime(ID string) time.Time {
+	m := []*updatecontrolmap.UpdateControlMap{}
+	query(c.Pool,
+		// Collector
+		func(u *updatecontrolmap.UpdateControlMap) {
+			m = append(m, u)
+		},
+		// Predicates
+		func(u *updatecontrolmap.UpdateControlMap) bool {
+			return u.ID == ID
+		},
+	)
+	// Return the time of the map with the closest time to halfway expired
+	if len(m) == 0 {
+		return time.Now() // TODO - fhwat to return (?)
```

Comment:
I would just return `UpdateControlMapExpirationTimeSeconds / 2`. This is guaranteed to be as long as, or shorter than, any map which is inserted later on.